File: Angular Code/src/app/Shared/employee.model.ts
import { EmployeeDependents } from "./employeedependent.model";

export class Employee {
    employeeId: number;
    firstName: string;
    lastName: string;
    salary: number;
    benefitsCostPerYear: number;
    benefitsCostPerPayCheck: number;
    dependentCount: number;
    dependentNames: string;
    displpayName: string;
    employeeDependents: EmployeeDependents[]

    constructor(firstName, lastName) {
        this.firstName = firstName;
        this.lastName = lastName;
    }
}

File: Dotnet API/PCTY/PCTY.BusinessService/Models/EmployeeModel.cs
using System.Collections.Generic;

namespace PCTY.BusinessService.Models
{
    public class EmployeeModel
    {
        public int EmployeeId { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public int Salary { get; set; }
        public bool IsActive { get; set; }
        public double BenefitsCostPerYear { get; set; }
        public double BenefitsCostPerPayCheck { get; set; }
        public string DisplpayName
        {
            get
            {
                return LastName + ", " + FirstName;
            }
        }

        public string DependentNames
        {
            get
            {
                if(EmployeeDependents != null)
                {
                    List<string> names = new List<string>();
                    foreach(var dependent in EmployeeDependents)
                    {
                        names.Add(dependent.LastName + ", " + dependent.FirstName);
                    }

                    return string.Join(" ; ", names);
                }

                return string.Empty;
            }
        }

        public int DependentCount
        {
            get
            {
                if (EmployeeDependents != null)
                {
                    return EmployeeDependents.Count;
                }

                return 0;
            }
        }

        public  IList<EmployeeDependentModel> EmployeeDependents { get; set; }
    }
}
